code/game/objects/items/devices/scanners.dm

```diff
@@ -693,12 +693,81 @@ GENE SCANNER
-	if (T.cores > 1)
+	if(T.cores > 1)
 		to_chat(user, "Multiple cores detected")
 	to_chat(user, "Growth progress: [T.amount_grown]/[SLIME_EVOLUTION_THRESHOLD]")
 	if(T.effectmod)
 		to_chat(user, "<span class='notice'>Core mutation in progress: [T.effectmod]</span>")
 		to_chat(user, "<span class = 'notice'>Progress in core mutation: [T.applied] / [SLIME_EXTRACT_CROSSING_REQUIRED]</span>")
+	if(T.transformeffects != SLIME_EFFECT_DEFAULT)
+		var/slimeeffects = "\nTransformative extract effects detected: "
+		var/i = T.effectsapplied - 1
+		if(T.transformeffects & SLIME_EFFECT_GREY)
+			slimeeffects += "grey[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_ORANGE)
+			slimeeffects += "orange[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_PURPLE)
+			slimeeffects += "purple[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_BLUE)
+			slimeeffects += "blue[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_METAL)
+			slimeeffects += "metal[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_YELLOW)
+			slimeeffects += "yellow[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_DARK_PURPLE)
+			slimeeffects += "dark purple[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_DARK_BLUE)
+			slimeeffects += "dark blue[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_SILVER)
+			slimeeffects += "silver[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_BLUESPACE)
+			slimeeffects += "bluespace[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_SEPIA)
+			slimeeffects += "sepia[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_CERULEAN)
+			slimeeffects += "cerulean[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_PYRITE)
+			slimeeffects += "pyrite[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_RED)
+			slimeeffects += "red[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_GREEN)
+			slimeeffects += "green[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_PINK)
+			slimeeffects += "pink[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_GOLD)
+			slimeeffects += "gold[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_OIL)
+			slimeeffects += "oil[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_BLACK)
+			slimeeffects += "black[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_LIGHT_PINK)
+			slimeeffects += "light pink[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_ADAMANTINE)
+			slimeeffects += "adamantine[(i == 0) ? "" : ", "]"
+			i--
+		if(T.transformeffects & SLIME_EFFECT_RAINBOW)
+			slimeeffects += "rainbow[(i == 0) ? "" : ", "]"
+		to_chat(user, "<span class='notice'>[slimeeffects]</span>")
```

Comment:
If there is a way of doing it without the if, if, if, if then that would be great.
If not remove i and have a list instead and add the slime effect name to the list each time then in the to_chat do list.join(",")
["blue", "pink"].join(", ") = "blue, pink"